vmdb/lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -88,9 +88,15 @@ def self.expose(*args)
         next if method_name.to_sym == :id
         self.association = method_name if options[:association]
         define_method(method_name) do |*params|
-          ret = object_send(options[:method] || method_name, *params)
+          method = options[:method] || method_name
+          ret = object_send(method, *params)
           return options[:override_return] if options.has_key?(:override_return)
-          return wrap_results(ret)
+          reflection = @object.class.reflection_with_virtual(method)
+          if reflection && reflection.collection?
```

Comment:
@Fryguy I agree, were you thinking of passing the reflection into the wrap_results method?  Is there a different way to get the `reflection.collection?` from the results itself?  It seems odd to pass the reflection since we're only processing the results in that method, not where the results came from.